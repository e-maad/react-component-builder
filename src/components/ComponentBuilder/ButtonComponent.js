import React from 'react';

const ButtonComponent = ({ text, id }) => {
    return <button id={id} className='button-component'>{text}</button>
}

export default ButtonComponent;