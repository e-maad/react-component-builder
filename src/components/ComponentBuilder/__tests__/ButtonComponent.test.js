import React from 'react';
import ButtonComponent from '../ButtonComponent';


const wrapper = shallow(<ButtonComponent />);
describe('Button Component', () => {

  it('match the snapshot', () => {
    const wrapper = renderer.create(<ButtonComponent />).toJSON();
    expect(wrapper).toMatchSnapshot();
  });

  it('base element should be present', () => {
    expect(wrapper.find('.button-component').length).toBe(1);
  });
});