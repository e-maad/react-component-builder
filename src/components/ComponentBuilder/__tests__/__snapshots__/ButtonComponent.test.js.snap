// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Component match the snapshot 1`] = `
<button
  className="button-component"
/>
`;
